pkg/networkdriver/cell.go

```diff
@@ -33,62 +36,26 @@ type networkDriverParams struct {
 	Lifecycle cell.Lifecycle
 	ClientSet k8sClient.Clientset
 	JobGroup  job.Group
+	Configs   resource.Resource[*v2alpha1.CiliumNetworkDriverConfig]
 }
 
-// getNetworkDriverConfig returns the network driver configuration.
-// This will be deprecated in favor a configuration passed via a Custom Resource.
-//
-// An example configuration to manage dummy devices is:
-//
-//	{
-//		DraRegistrationRetry:   time.Second,
-//		DraRegistrationTimeout: 30 * time.Second,
-//		PublishInterval:        3 * time.Second,
-//		DriverName:             "dummy.cilium.k8s.io",
-//		DeviceManagerConfigs: map[types.DeviceManagerType]types.DeviceManagerConfig{
-//			types.DeviceManagerTypeDummy: dummy.DummyConfig{Enabled: true},
-//		},
-//		Pools: []PoolConfig{
-//			{
-//				Name: "dummy-devices",
-//				Filter: types.DeviceFilter{
-//					DriverTypes: []types.DeviceManagerType{types.DeviceManagerTypeDummy},
-//					IfNames:     []string{"dummy"},
-//				},
-//			},
-//		},
-//	}
-func getNetworkDriverConfig(_ k8sClient.Clientset) (*v2alpha1.CiliumNetworkDriverConfigSpec, error) {
-	return nil, nil
+func getNetworkDriverConfigResource(cs k8sClient.Clientset, lc cell.Lifecycle, mp workqueue.MetricsProvider) resource.Resource[*v2alpha1.CiliumNetworkDriverConfig] {
```

Comment:
Nit: conventionally we name this functions `<resource_name>Resource`, that is `ciliumNetworkDriverConfigResource` in this case